clab/clab.go

```diff
@@ -511,60 +519,43 @@ func (c *CLab) WaitForExternalNodeDependencies(ctx context.Context, nodeName str
 }
 
 // CreateLinks creates links using the specified number of workers.
-func (c *CLab) CreateLinks(ctx context.Context, workers uint) {
+func (c *CLab) CreateLinks(ctx context.Context, workers uint, dm dependency_manager.DependencyManager) {
 	wg := new(sync.WaitGroup)
-	wg.Add(int(workers))
-	linksChan := make(chan *types.Link)
+	sem := semaphore.NewWeighted(int64(workers))
 
-	log.Debug("creating links...")
-	// wire the links between the nodes based on cabling plan
-	for i := uint(0); i < workers; i++ {
-		go func(i uint) {
+	for _, link := range c.Links {
+		wg.Add(1)
+		go func(li *types.Link) {
 			defer wg.Done()
-			for {
-				select {
-				case link := <-linksChan:
-					if link == nil {
-						log.Debugf("Link worker %d terminating...", i)
-						return
-					}
-					log.Debugf("Link worker %d received link: %+v", i, link)
-					if err := c.CreateVirtualWiring(link); err != nil {
-						log.Error(err)
-					}
-				case <-ctx.Done():
-					return
+
+			var waitNodes []string
+			for _, n := range []*types.NodeConfig{li.A.Node, li.B.Node} {
+				// we should not wait for "host"
+				if n.Kind != "host" {
```

Comment:
@steiler I think this should be also `ovs` and `ext-cont`?
I also wonder, maybe you want this to enhance so that each node signal SignalDone to the dep_manager so that you don't have to hard code the kinds?

Reply:
Well I'm not 100% sure to be hones. But I think no, this should not be needed for `bridge`, `ovs` and `ext-cont`.
All of these kinds actually have their Node implementation and will therefore be "deployed" and a Done is being signaled.
In general, this also holds true for host, we have a Node for that.
But this is where the whole stuff is all mixed and messed up.
We have to differentiate between a node definition of kind `host` and a link endpoint that states `host` in the node part.
They can both exist but don't have to.